lib/dicom_net/association.ex

```diff
@@ -158,9 +158,24 @@ defmodule DicomNet.Association do
     # TODO this is not always c-store
 
     ds = Dicom.BinaryFormat.from_binary(data, ts_options)
-    msg = {:dicom, %{operation: :cstore, dataset: ds}}
-    send(event_listener, msg)
-    response_ds = handle_cstore(command, ds)
+    commandField = command[256].values 
```

Comment:
I would stay with using `Dataset.fetch!(:Keyword)`. That's much easier to read and doesn't force DataSet to always remain a simple map.